src/components/dropdown/dropdown.js

```diff
@@ -0,0 +1,147 @@
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import Icon from '../icon';
+import Loading from '../loading';
+
+import dropdownArrow from './assets/dropdown-arrow.svg';
+import infoIcon from './assets/info.svg';
+
+import selectize from '../../styles/react-selectize.scss';
+import styles from './dropdown-styles.scss';
+
+class Dropdown extends PureComponent {
+  componentDidUpdate() {
+    this.selectorElement.highlightFirstSelectableOption();
+  }
+
+  render() {
+    const {
+      loading,
+      white,
+      label,
+      transparent,
+      plain,
+      dark,
+      blueBorder,
+      theme,
+      disabled,
+      colorDot,
+      info,
+      infoText,
+      required,
+      optional,
+      value
+    } = this.props;
+    const themeIcon = white ? { icon: styles.white } : {};
+    const hasNotValue = value && !value.value;
+    const isRequired = hasNotValue && required;
+    return (
+      <div
+        className={cx(
+          selectize.dropdown,
+          styles.dropdownWrapper,
+          {
+            [styles.flex]: colorDot,
+            [styles.requiredError]: required && hasNotValue
+          },
+          theme.wrapper
+        )}
+      >
+        {
+          colorDot &&
+            (
+              <span
+                className={styles.dot}
+                style={{ backgroundColor: colorDot }}
+              />
+            )
+        }
+        {label && <span className={styles.label}>{label}</span>}
+        {
+          isRequired &&
+            <span className={styles.requiredError}>This field is required</span>
+        }
+        {optional && <span className={styles.optional}>(optional)</span>}
+        {
+          info && (
+          <div data-tip={infoText} className={styles.infoContainer}>
+            <Icon icon={infoIcon} className={styles.infoIcon} />
+          </div>
+            )
+        }
+        <div
+          className={cx(
+            styles.dropdown,
+            transparent ? styles.transparent : '',
+            white ? styles.white : '',
+            plain ? styles.plain : '',
+            dark ? styles.dark : '',
+            blueBorder ? styles.blueBorder : ''
+          )}
+        >
+          {loading && <Loading className={styles.loader} mini />}
+          <SimpleSelect
+            ref={el => {
+              this.selectorElement = el;
+            }}
+            className={cx(theme.select, disabled, {
+              [styles.withDot]: colorDot
+            })}
+            renderToggleButton={() => (
+              <Icon icon={dropdownArrow} theme={themeIcon} />
+            )}
+            {...this.props}
+          />
+        </div>
+      </div>
+    );
+  }
+}
+
+Dropdown.propTypes = {
+  label: PropTypes.string,
+  openUp: PropTypes.bool,
+  transparent: PropTypes.bool,
+  white: PropTypes.bool,
+  plain: PropTypes.bool,
+  dark: PropTypes.bool,
+  info: PropTypes.bool,
+  infoText: PropTypes.string,
+  theme: PropTypes.object,
```

Comment:
I think we should be consistent and keep using `PropTypes.shape({...})` for defining what properties theme may have.

Overall, about the component; I know it's extracted but it just seems too crowded; following your proposition from yesterday about extensive styling options, maybe we should get rid of some? I would suggest getting rid of `white`, `plain`, `dark`, `blueBorder` and `colorDot` and replace them with theming, wdyt?

Reply:
Agreed! I forgot to remove them thanks!
I'll move to the theme to keep the original functionality as examples.

Reply:
Removed all of them except the withDot as it embed the feature of including the `dot` so renamed to withDot boolean and theme responsability to update the color.